libs/langchain/langchain/agents/openai_functions_agent/base.py

```diff
@@ -226,3 +227,120 @@ def from_llm_and_tools(
             callback_manager=callback_manager,
             **kwargs,
         )
+
+
+def create_default_prompt(
+    system_message: Optional[str] = "You are a helpful AI assistant.",
+    include_chat_history: bool = False,
+):
+    """Create default prompt for OpenAI Functions agent.
+
+    Args:
+        system_message: String to use as the system message,
+            defaults to "You are a helpful AI assistant."
+        include_chat_history: Whether to include a variable for chat history or not.
+            If yes, this will require an input of `chat_history`.
+
+    Returns:
+        A prompt template. Requires two or three variables:
+        - `input`: this is the user input
+        - `agent_scratchpad`: this is the history of steps the agent has taken so far.
+            This gets populated by the AgentExecutor.
+        - `chat_history`: This is optionally required as input, if
+            `include_chat_history` is set to True. It should be a list of LangChain
+            Message objects.
+    """
+    messages: List[Union[BaseMessagePromptTemplate, BaseMessage]]
+    messages = [SystemMessage(content=system_message)]
+    if include_chat_history:
+        messages.append(MessagesPlaceholder(variable_name="chat_history"))
+    messages.extend(
+        [
+            HumanMessagePromptTemplate.from_template("{input}"),
+            MessagesPlaceholder(variable_name="agent_scratchpad"),
+        ]
+    )
+    return ChatPromptTemplate(messages=messages)
+
+
+def create_openai_functions_agent(
```

Comment:
Worth renaming to `create_functions_agent` and having typing be some kind of `FunctionCallingChatModel` since you could do this with Anthropic/potentially Fireworks/Ollama?